test/compilationTests/stringutils/strings.sol

```diff
@@ -210,8 +211,11 @@ library strings {
             }
             if (a != b) {
                 // Mask out irrelevant bytes and check again
-                uint mask = ~(2 ** (8 * (32 - shortest + idx)) - 1);
-                uint diff = (a & mask) - (b & mask);
+                uint256 mask = uint256(-1); // 0xffff...
+                if(shortest < 32) {
```

Comment:
Why is the old mask now guarded?

Reply:
I'm not sure, probably due to the fact `short >= 32` will likely have a negative exponent?

I don't think we need to take care of this at all though, it is only compilation tests, and it is taken vanilla from upstream.